dev-tools/run_with_go_ver

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+#
+# This sets up Go based on the project's Go version. It will configure
+# GOROOT and add GOROOT/bin to PATH before executing the given command.
+#
+# Example usage: ./run_with_go_ver go version
+#
+set -e
+
+# Script directory:
+SDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+source "${SDIR}/common.bash"
+
+get_go_version
+setup_go_root ${GO_VERSION}
+bash -c "$*"
```

Comment:
I think the CI workers already have some macros to do things like this, for example I think this is implemented by `withGoEnv` as defined [here](https://github.com/elastic/apm-pipeline-library/blob/main/vars/withGoEnv.groovy).

This would be used similarly to `withMageEnv` here: https://github.com/elastic/elastic-agent-shipper/blob/cb5cf808608678de86f85d353ab7fb1c41260465/.ci/Jenkinsfile#L41

This is all assuming we want this to run on the Jenkins workers, and not somewhere else that doesn't have these features. I just want to double check to prevent us from maintaining two versions of something :)

Reply:
yes the idea is to have the option to run this everywhere.